presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetCompressionUtils.java

```diff
@@ -86,29 +86,27 @@ private static Slice decompressGzip(Slice input, int uncompressedSize)
         }
     }
 
+    // decompress LZO-compressed data using HADOOP-LZO codec
     private static Slice decompressLZO(Slice input, int uncompressedSize)
+            throws IOException
     {
-        LzoDecompressor lzoDecompressor = new LzoDecompressor();
-        long totalDecompressedCount = 0;
-        // over allocate buffer which makes decompression easier
-        byte[] output = new byte[uncompressedSize + SIZE_OF_LONG];
-        int outputOffset = 0;
-        int inputOffset = 0;
-        int cumulativeUncompressedBlockLength = 0;
-
-        while (totalDecompressedCount < uncompressedSize) {
-            if (totalDecompressedCount == cumulativeUncompressedBlockLength) {
-                cumulativeUncompressedBlockLength += Integer.reverseBytes(input.getInt(inputOffset));
-                inputOffset += SIZE_OF_INT;
-            }
-            int compressedChunkLength = Integer.reverseBytes(input.getInt(inputOffset));
-            inputOffset += SIZE_OF_INT;
-            int decompressionSize = decompress(lzoDecompressor, input, inputOffset, compressedChunkLength, output, outputOffset);
-            totalDecompressedCount += decompressionSize;
-            outputOffset += decompressionSize;
-            inputOffset += compressedChunkLength;
+        byte[] output = new byte[uncompressedSize];
+        LzoCodec codec = new LzoCodec();
+        codec.setConf(new Configuration());
+
+        // input data
+        byte[] byteArray = (byte[]) input.getBase();
+        int inputOffset = (int) input.getAddress() - ARRAY_BYTE_BASE_OFFSET;
+
+        InputStream is = new ByteArrayInputStream(byteArray, inputOffset, input.length());
+        is = codec.createInputStream(is);
+
+        int decompressionSize = 0;
+        while (decompressionSize < uncompressedSize) {
+            decompressionSize += is.read(output, decompressionSize, uncompressedSize - decompressionSize);
```

Comment:
How would it be difficult to manage the byte array offset by presto to get rid of InputStream instance which is copying buffer? I imagine it shouldn't be really hard and should just a few lines to read block size and `setInput` for the decompressor. If it does have too much work, we can leave a comment and address later.